### PR TITLE
[python.d/samba] Only use sudo when not running as root user

### PR DESCRIPTION
**Pull request by Duffyx**
##### Summary
Only use sudo when not running as root user

##### Component Name
python.d samba collector

##### Test Plan
I have applied this patch to my instance and all is good so far monitoring samba.

### CODE REVIEW

**Comment by ktsaou** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- collectors/python.d.plugin/samba/samba.chart.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/netdata/netdata/pullRequest?prid=5524579&bid=18102385)

**Comment by ktsaou** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- collectors/python.d.plugin/samba/samba.chart.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/netdata/netdata/pullRequest?prid=5524579&bid=18102385)

File: collectors/python.d.plugin/samba/samba.chart.py [hunk 1]
```diff
@@ -107,25 +108,27 @@ def __init__(self, configuration=None, name=None):
...
+            smbstatus = '{0} -P'.format(smbstatus_binary)
+            allowed = self._get_raw_data(command=command)
+            if not (allowed and allowed[0].strip() == smbstatus):
+                self.error("not allowed to run sudo for command '{0}'".format(smbstatus))
+                return False
+            self.command = ' '.join([sudo_binary, '-n', smbstatus_binary, '-P'])
+            return ExecutableService.check(self)
+        else:
+            self.command = ' '.join([smbstatus_binary, '-P'])
+            return ExecutableService.check(self)
```
Comment (ilyam8) [inline collectors/python.d.plugin/samba/samba.chart.py:131]: i suggest to not use `else` 😄 it is easier to read imo 

```python
if os.getuid() == 0:
  ...

...
```

**Comment by ktsaou** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- collectors/python.d.plugin/samba/samba.chart.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/netdata/netdata/pullRequest?prid=5524579&bid=18102385)

File: collectors/python.d.plugin/samba/samba.chart.py (hunk 1 above)
Comment (Duffyx) [reply collectors/python.d.plugin/samba/samba.chart.py:118]: This should be resolved now :)

File: collectors/python.d.plugin/samba/samba.chart.py [hunk 2]
```diff
@@ -107,23 +108,25 @@ def __init__(self, configuration=None, name=None):
...
-        if not sudo_binary:
-            self.error("can't locate '{0}' binary".format(SUDO))
-            return False
-
         smbstatus_binary = find_binary(SMBSTATUS)
         if not smbstatus_binary:
             self.error("can't locate '{0}' binary".format(SMBSTATUS))
             return False
 
+        if os.getuid() == 0:
```
Comment (ilyam8) [inline collectors/python.d.plugin/samba/samba.chart.py:116]: this should be in the beginning, there is no need to check for `smbstatus_binary` if uid == 0

Comment (ilyam8) [reply collectors/python.d.plugin/samba/samba.chart.py:116]: 🙈

**Comment by ilyam8** [comment]
thx for the conrtibution @Duffyx

**Comment by ilyam8** [comment]
> Not yet done anything to test

The changes look straightforward, i am sure they break nothing but do you plan to test them?

**Comment by Duffyx** [comment]
> > Not yet done anything to test
> 
> The changes look straightforward, i am sure they break nothing but do you plan to test them?

I have just applied these changes to my netdata instance and all is good :-)

**Pull request merged**